services/qs-spark-service/src/main/java/ai/quantumics/api/service/impl/RuleJobServiceImpl.java

```diff
@@ -96,40 +99,32 @@ public ResponseEntity<Object> runRuleJob(RuleJobRequest ruleJobRequest, int user
             }
 
             dbUtil.changeSchema(project.getDbSchemaName());
+            List<String> statuses = Arrays.asList(RuleJobStatus.INPROCESS.getStatus(), RuleJobStatus.NOT_STARTED.getStatus());
             for (Integer ruleId : ruleJobRequest.getRuleIds()) {
                 QsRule rule = ruleRepository.findByRuleId(ruleId);
-                QsRuleJob ruleJob = ruleJobRepository.findByRuleIdAndActiveIsTrue(ruleId);
-                if (ruleJob != null && (ruleJob.getJobStatus().equals(RuleJobStatus.INPROCESS.getStatus()) || ruleJob.getJobStatus().equals(RuleJobStatus.NOT_STARTED.getStatus()))) {
-					inProcessRules.add(ruleId);
+                List<QsRuleJob> ruleJobs = ruleJobRepository.findByRuleIdAndActiveIsTrueAndJobStatusInAndBusinessDate(ruleId, statuses, LocalDate.now());
```

Comment:
1.I think, we should have check of current number of jobs are currently inprogress.. until then new jobs are not allowed .
2. It will be always localDate (current date)?

Reply:
1. I did not understood this point let's connect tomorrow on this.
2. Yes for pilot. No for pattern changes. Once we will implement for pattern then we will check for the date which are coming in patter.